src/common/ui-wrapper.js

```diff
@@ -55,13 +55,30 @@ class UIWrapper {
   /**
    * Add a component dynamically
    *
-   * @param {KPUIComponent} component - The component to add
+   * @param {KPUIAddComponent} component - The component to add
    * @returns {Function} - Removal function
    */
-  addComponent(component: KPUIComponent): Function {
+  addComponent(component: KPUIAddComponent): Function {
     return this._uiManager.addComponent(component);
   }
 
+  /**
+   * Remove a component dynamically
+   *
+   * @param {KPUIRemoveComponent} component - The component to remove
+   * @returns {Function} - Undo removal function
+   */
+  removeComponent(component: KPUIRemoveComponent): Function {
+    let replaceComponent: KPUIAddComponent = {
+      label: `Remove_${component.removeComponent}`,
+      get: Components.Remove,
+      presets: component.presets.concat(),
```

Comment:
why concat is needed?

Reply:
To create a copy of the array

Reply:
1. why copy?
2. i think `[...component.presets]` is clearer

Reply:
1.Copy because it seemed to me more correct. But not sure actually why not.
2. Because I'm old and I wrote it 1999 style